<a href="https://colab.research.google.com/github/Randasabag/Etude-de-march-de-la-poule-qui-chante-avec-Python/blob/main/projet9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <font color='indianred'> LA POULE QUI CHANTE </font>
##MISSION : Analyse pour le lancement à l’international en groupant les pays à cibler pour exporter nos poulets

1ère partie : Prétraitement des données


2ème partie : Analyse  
<li>ACP afin de visualiser les résultats de ton analyse, comprendre les groupes, les liens entre les variables, les liens entre les individus</li>
<li>Méthode des k-means, afin d’affiner l’analyse et comparer les résultats des deux méthodes de clustering et analyse des centroides des classes</li>
<li>Classification ascendante hiérarchique avec un dendrogramme comme visualisation</li>


# Sommaire :
**Partie 1 : Importation des données**
 - <a href="#C1">Importation des librairies</a>
 - <a href="#C2">Importation des jeux de données</a>
 
**Partie 2 : Analyse des jeux de données**
 - <a href="#C3">Affichage du nombre de lignes et colonnes des jeux de données </a>
 - <a href="#C4">Affichage des informations des jeux de données </a>
 - <a href="#C5">Analyse des valeurs manquantes</a>
 - <a href="#C6">Unicité de la clé primaire</a>
 - <a href="#C7">Analyse des doublons</a>
 - <a href="#C8">Jointure des tables</a>
 - <a href="#C9">Indicateurs choisis</a>



**Partie 3 : Analyse des indicateurs**


# **<font color='indianred'>Partie 1 : Importation des données</font>**

# <a name="C1"><font color='LightSeaGreen'>Importation des librairies</font></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import io
from google.colab import files

# <a name="C2"><font color='LightSeaGreen'>Importation des données</font></a>

In [ ]:
uploaded = files.upload()

Saving DisponibiliteAlimentaire_2017.csv to DisponibiliteAlimentaire_2017.csv
Saving pibindice.csv to pibindice.csv
Saving Population_2000_2018.csv to Population_2000_2018.csv


In [ ]:
#Disponibilité alimentaire pour l'année 2017
df_dispo=pd.read_csv(io.BytesIO(uploaded['DisponibiliteAlimentaire_2017.csv'])) 
df_dispo.head()

Code Domaine                      Domaine  Code zone         Zone  \
0          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
1          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
2          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
3          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
4          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   

   Code Élément                   Élément  Code Produit          Produit  \
0          5511                Production          2511  Blé et produits   
1          5611   Importations - Quantité          2511  Blé et produits   
2          5072        Variation de stock          2511  Blé et produits   
3          5911   Exportations - Quantité          2511  Blé et produits   
4          5301  Disponibilité intérieure          2511  Blé et produits   

   Code année  Année               Unité  Valeur Symbole  \
0        2017   2017  Milliers de tonnes  4281.0       S   
1        2017   2017  Milliers de tonnes  2302.0       S   
2        2017   2017  Milliers de tonnes  -119.0       S   
3        2017   2017  Milliers de tonnes     0.0       S   
4        2017   2017  Milliers de tonnes  6701.0       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées  
3  Données standardisées  
4  Données standardisées

In [ ]:
#Population entre les années 2000 et 2018
df_pop=pd.read_csv(io.BytesIO(uploaded['Population_2000_2018.csv'])) 
df_pop.head()

Code Domaine                       Domaine  Code zone         Zone  \
0           OA  Séries temporelles annuelles          2  Afghanistan   
1           OA  Séries temporelles annuelles          2  Afghanistan   
2           OA  Séries temporelles annuelles          2  Afghanistan   
3           OA  Séries temporelles annuelles          2  Afghanistan   
4           OA  Séries temporelles annuelles          2  Afghanistan   

   Code Élément            Élément  Code Produit                 Produit  \
0           511  Population totale          3010  Population-Estimations   
1           511  Population totale          3010  Population-Estimations   
2           511  Population totale          3010  Population-Estimations   
3           511  Population totale          3010  Population-Estimations   
4           511  Population totale          3010  Population-Estimations   

   Code année  Année           Unité     Valeur Symbole  \
0        2000   2000  1000 personnes  20779.953       X   
1        2001   2001  1000 personnes  21606.988       X   
2        2002   2002  1000 personnes  22600.770       X   
3        2003   2003  1000 personnes  23680.871       X   
4        2004   2004  1000 personnes  24726.684       X   

          Description du Symbole Note  
0  Sources internationales sûres  NaN  
1  Sources internationales sûres  NaN  
2  Sources internationales sûres  NaN  
3  Sources internationales sûres  NaN  
4  Sources internationales sûres  NaN

In [ ]:
#Tableau regroupant les données sur le PIB/habitant et l'indice de stabilité politique
df_pibind=pd.read_csv(io.BytesIO(uploaded['pibindice.csv'])) 
df_pibind.head()

Code Domaine                             Domaine  Code zone (FAO)  \
0           FS  Données de la sécurité alimentaire                2   
1           FS  Données de la sécurité alimentaire                2   
2           FS  Données de la sécurité alimentaire              202   
3           FS  Données de la sécurité alimentaire              202   
4           FS  Données de la sécurité alimentaire                3   

             Zone  Code Élément Élément  Code Produit  \
0     Afghanistan          6126  Valeur         22013   
1     Afghanistan          6125  Valeur         21032   
2  Afrique du Sud          6126  Valeur         22013   
3  Afrique du Sud          6125  Valeur         21032   
4         Albanie          6126  Valeur         22013   

                                             Produit  Code année  Année  \
0  PIB par habitant, ($ PPA internationaux consta...        2017   2017   
1  Stabilité politique et absence de violence/ter...        2017   2017   
2  PIB par habitant, ($ PPA internationaux consta...        2017   2017   
3  Stabilité politique et absence de violence/ter...        2017   2017   
4  PIB par habitant, ($ PPA internationaux consta...        2017   2017   

    Unité    Valeur Symbole         Description du Symbole  Note  
0      I$   2058.40       X  Sources internationales sûres   NaN  
1  indice     -2.80       X  Sources internationales sûres   NaN  
2      I$  12701.30       X  Sources internationales sûres   NaN  
3  indice     -0.28       X  Sources internationales sûres   NaN  
4      I$  12771.00       X  Sources internationales sûres   NaN

##<font color='purple'>Préparation des données</font>

### Table Disponibilité Alimentaire

In [ ]:
#La table Disponibilité Alimentaire contient des données pour l'année 2017
#Garder la viande de volailles 
df_dispovol=df_dispo[df_dispo['Produit']=='Viande de Volailles']     
#renomme la colonne zone par pays
df_dispovol=df_dispovol.rename(columns={'Zone':'Pays'})   
df_dispovol.head()

Code Domaine                      Domaine  Code zone         Pays  \
651          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
652          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
653          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
654          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
655          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   

     Code Élément                   Élément  Code Produit  \
651          5511                Production          2734   
652          5611   Importations - Quantité          2734   
653          5072        Variation de stock          2734   
654          5301  Disponibilité intérieure          2734   
655          5123                    Pertes          2734   

                 Produit  Code année  Année               Unité  Valeur  \
651  Viande de Volailles        2017   2017  Milliers de tonnes    28.0   
652  Viande de Volailles        2017   2017  Milliers de tonnes    29.0   
653  Viande de Volailles        2017   2017  Milliers de tonnes     0.0   
654  Viande de Volailles        2017   2017  Milliers de tonnes    57.0   
655  Viande de Volailles        2017   2017  Milliers de tonnes     2.0   

    Symbole Description du Symbole  
651       S  Données standardisées  
652       S  Données standardisées  
653       S  Données standardisées  
654       S  Données standardisées  
655       S  Données standardisées

In [ ]:
df_dispovol=df_dispovol[['Pays', 'Élément', 'Unité', 'Valeur']]
df_dispovol.head()

Pays                   Élément               Unité  Valeur
651  Afghanistan                Production  Milliers de tonnes    28.0
652  Afghanistan   Importations - Quantité  Milliers de tonnes    29.0
653  Afghanistan        Variation de stock  Milliers de tonnes     0.0
654  Afghanistan  Disponibilité intérieure  Milliers de tonnes    57.0
655  Afghanistan                    Pertes  Milliers de tonnes     2.0

In [ ]:
df_dispovol=df_dispovol.pivot_table(index='Pays', columns='Élément', values='Valeur', aggfunc=sum)
df_dispovol.head()

Élément         Alimentation pour touristes  Aliments pour animaux  \
Pays                                                                 
Afghanistan                             NaN                    NaN   
Afrique du Sud                          0.0                    NaN   
Albanie                                 NaN                    NaN   
Algérie                                 0.0                    NaN   
Allemagne                               NaN                    NaN   

Élément         Autres utilisations (non alimentaire)  \
Pays                                                    
Afghanistan                                       NaN   
Afrique du Sud                                    NaN   
Albanie                                           NaN   
Algérie                                           NaN   
Allemagne                                         NaN   

Élément         Disponibilité alimentaire (Kcal/personne/jour)  \
Pays                                                             
Afghanistan                                                5.0   
Afrique du Sud                                           143.0   
Albanie                                                   85.0   
Algérie                                                   22.0   
Allemagne                                                 71.0   

Élément         Disponibilité alimentaire en quantité (kg/personne/an)  \
Pays                                                                     
Afghanistan                                                  1.53        
Afrique du Sud                                              35.69        
Albanie                                                     16.36        
Algérie                                                      6.38        
Allemagne                                                   19.47        

Élément         Disponibilité de matière grasse en quantité (g/personne/jour)  \
Pays                                                                            
Afghanistan                                                  0.33               
Afrique du Sud                                               9.25               
Albanie                                                      6.45               
Algérie                                                      1.50               
Allemagne                                                    4.16               

Élément         Disponibilité de protéines en quantité (g/personne/jour)  \
Pays                                                                       
Afghanistan                                                  0.54          
Afrique du Sud                                              14.11          
Albanie                                                      6.26          
Algérie                                                      1.97          
Allemagne                                                    7.96          

Élément         Disponibilité intérieure  Exportations - Quantité  \
Pays                                                                
Afghanistan                         57.0                      NaN   
Afrique du Sud                    2118.0                     63.0   
Albanie                             47.0                      0.0   
Algérie                            277.0                      0.0   
Allemagne                         1739.0                    646.0   

Élément         Importations - Quantité  Nourriture  Pertes  Production  \
Pays                                                                      
Afghanistan                        29.0        55.0     2.0        28.0   
Afrique du Sud                    514.0      2035.0    83.0      1667.0   
Albanie                            38.0        47.0     NaN        13.0   
Algérie                             2.0       264.0    13.0       275.0   
Allemagne                         842.0      1609.0     NaN      1514.0   

Élément         

In [ ]:
df_dispovol=df_dispovol.reset_index()
df_dispovol=df_dispovol[['Pays', 'Disponibilité alimentaire en quantité (kg/personne/an)', 'Disponibilité intérieure', 'Exportations - Quantité',	'Importations - Quantité', 'Production', 'Variation de stock']]
df_dispovol.head(2)

Élément            Pays  \
0           Afghanistan   
1        Afrique du Sud   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
1                                                    35.69        

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                            57.0                      NaN   
1                          2118.0                     63.0   

Élément  Importations - Quantité  Production  Variation de stock  
0                           29.0        28.0                 0.0  
1                          514.0      1667.0                 0.0

### Table Population

In [ ]:
#Garder l'année 2017 
df_pop2017=df_pop[df_pop['Année']==2017]        
df_pop2017.head()

Code Domaine                       Domaine  Code zone            Zone  \
17           OA  Séries temporelles annuelles          2     Afghanistan   
36           OA  Séries temporelles annuelles        202  Afrique du Sud   
55           OA  Séries temporelles annuelles          3         Albanie   
74           OA  Séries temporelles annuelles          4         Algérie   
93           OA  Séries temporelles annuelles         79       Allemagne   

    Code Élément            Élément  Code Produit                 Produit  \
17           511  Population totale          3010  Population-Estimations   
36           511  Population totale          3010  Population-Estimations   
55           511  Population totale          3010  Population-Estimations   
74           511  Population totale          3010  Population-Estimations   
93           511  Population totale          3010  Population-Estimations   

    Code année  Année           Unité     Valeur Symbole  \
17        2017   2017  1000 personnes  36296.113       X   
36        2017   2017  1000 personnes  57009.756       X   
55        2017   2017  1000 personnes   2884.169       X   
74        2017   2017  1000 personnes  41389.189       X   
93        2017   2017  1000 personnes  82658.409       X   

           Description du Symbole Note  
17  Sources internationales sûres  NaN  
36  Sources internationales sûres  NaN  
55  Sources internationales sûres  NaN  
74  Sources internationales sûres  NaN  
93  Sources internationales sûres  NaN

In [ ]:
df_pop2017=df_pop2017[['Zone', 'Unité', 'Valeur']]
df_pop2017.head()

Zone           Unité     Valeur
17     Afghanistan  1000 personnes  36296.113
36  Afrique du Sud  1000 personnes  57009.756
55         Albanie  1000 personnes   2884.169
74         Algérie  1000 personnes  41389.189
93       Allemagne  1000 personnes  82658.409

In [ ]:
#renomme la colonne zone par pays
df_pop2017=df_pop2017.rename(columns={'Zone':'Pays', 'Valeur':'Population'}) 
df_pop2017.head()

Pays           Unité  Population
17     Afghanistan  1000 personnes   36296.113
36  Afrique du Sud  1000 personnes   57009.756
55         Albanie  1000 personnes    2884.169
74         Algérie  1000 personnes   41389.189
93       Allemagne  1000 personnes   82658.409

In [ ]:
df_pop2017['Population']=df_pop2017['Population'] * 1000
df_pop2017.head()

Pays           Unité  Population
17                               Afghanistan  1000 personnes  36296113.0
36                            Afrique du Sud  1000 personnes  57009756.0
55                                   Albanie  1000 personnes   2884169.0
74                                   Algérie  1000 personnes  41389189.0
93                                 Allemagne  1000 personnes  82658409.0
...                                      ...             ...         ...
4333  Venezuela (République bolivarienne du)  1000 personnes  29402484.0
4352                                Viet Nam  1000 personnes  94600648.0
4371                                   Yémen  1000 personnes  27834819.0
4390                                  Zambie  1000 personnes  16853599.0
4409                                Zimbabwe  1000 personnes  14236595.0

[236 rows x 3 columns]

In [ ]:
df_pop2017=df_pop2017[['Pays', 'Population']]
df_pop2017.head()

Pays  Population
17                               Afghanistan  36296113.0
36                            Afrique du Sud  57009756.0
55                                   Albanie   2884169.0
74                                   Algérie  41389189.0
93                                 Allemagne  82658409.0
...                                      ...         ...
4333  Venezuela (République bolivarienne du)  29402484.0
4352                                Viet Nam  94600648.0
4371                                   Yémen  27834819.0
4390                                  Zambie  16853599.0
4409                                Zimbabwe  14236595.0

[236 rows x 2 columns]

### Table PIB et Indice de stabilité politique

In [ ]:
#df_pibind.head(2)

In [ ]:
df_pibind=df_pibind.pivot_table(index='Zone', columns='Produit', values='Valeur', aggfunc=sum)
df_pibind.head()

Produit         PIB par habitant, ($ PPA internationaux constants de 2011)  \
Zone                                                                         
Afghanistan                                                2058.4            
Afrique du Sud                                            12701.3            
Albanie                                                   12771.0            
Algérie                                                   11737.4            
Allemagne                                                 52952.9            

Produit         Stabilité politique et absence de violence/terrorisme (indice)  
Zone                                                                            
Afghanistan                                                 -2.80               
Afrique du Sud                                              -0.28               
Albanie                                                      0.38               
Algérie                                                     -0.92               
Allemagne                                                    0.59

In [ ]:
df_pibind=df_pibind.reset_index()
#renomme la colonne zone par pays
df_pibind=df_pibind.rename(columns={'Zone':'Pays'}) 
df_pibind.head()

Produit            Pays  \
0           Afghanistan   
1        Afrique du Sud   
2               Albanie   
3               Algérie   
4             Allemagne   

Produit  PIB par habitant, ($ PPA internationaux constants de 2011)  \
0                                                   2058.4            
1                                                  12701.3            
2                                                  12771.0            
3                                                  11737.4            
4                                                  52952.9            

Produit  Stabilité politique et absence de violence/terrorisme (indice)  
0                                                    -2.80               
1                                                    -0.28               
2                                                     0.38               
3                                                    -0.92               
4                                                     0.59

# **<font color='indianred'>Partie 2 : Analyse des jeux de données</font>**

# <a name="C3"><font color='LightSeaGreen'>Affichage du nombre de lignes et colonnes des jeux de données</font></a>

In [ ]:
#nombre de lignes et colonnes
print('Disponibilité alimentaire',df_dispovol.shape)
print('Population',df_pop2017.shape)
print('PIB_Indice',df_pibind.shape)

Disponibilité alimentaire (172, 7)
Population (236, 2)
PIB_Indice (198, 3)


# <a name="C4"><font color='LightSeaGreen'>Affichage des informations des jeux de données</font></a>

In [ ]:
print('Table Disponibilité Alimentaire')
df_dispovol.info()
print('\n')
print('Table Population')
print('-------------------------------------')
df_pop2017.info()
print('\n')
print('Table PIB Indice')
print('-------------------------------------')
df_pibind.info()

Table Disponibilité Alimentaire
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 172 entries, 0 to 171
Data columns (total 7 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   Pays                                                    172 non-null    object 
 1   Disponibilité alimentaire en quantité (kg/personne/an)  172 non-null    float64
 2   Disponibilité intérieure                                170 non-null    float64
 3   Exportations - Quantité                                 135 non-null    float64
 4   Importations - Quantité                                 170 non-null    float64
 5   Production                                              168 non-null    float64
 6   Variation de stock                                      169 non-null    float64
dtypes: float64(6), object(1)
memory usage: 9.5+ KB


Table Population
----------------------------

# <a name="C5"><font color='LightSeaGreen'>Analyse des valeurs manquantes</font></a>

In [ ]:
#Création d'une fonction calculant le pourcentage de valeurs manquantes 
#et affichant une heatmap pour visualiser les valeurs manquantes 
def val_manq(df):
    ligne,colonne=df.shape
    nb_cell=ligne*colonne
    nb_null=df.isnull().sum().sum()
    prct=nb_null *100/nb_cell
    print('Nombre de valeurs manquantes =',nb_null)
    print('Nombre de cellulles totales =',nb_cell)
    print('Pourcentage de valeurs manquantes = ',round(prct,2),'%') 
    print('\n')
    #print('Pourcentage de valeurs manquantes avec le nombre de colonnes =\n',(df.isna().sum()).value_counts(ascending=True))
    #Calcul des pourcentages de valeurs manquantes par variable(colonne)
    print('\n')
    print('Pourcentage de valeurs manquantes par variable/colonne =\n', 
          (df.isna().sum()).sort_values(ascending=True))
    print('\n')

In [ ]:
#Appel de la fonction val_manq 
#Affichage des pourcentages de valeurs manquantes des 2 jeux de données 
print('1-Table disponibilité alimentaire :')
val_manq(df_dispovol)
print('---------------------------------------')
print('2-Table population :')
val_manq(df_pop2017)
print('---------------------------------------')
print('3-Table PIB Indice :')
val_manq(df_pibind)

1-Table disponibilité alimentaire :
Nombre de valeurs manquantes = 48
Nombre de cellulles totales = 1204
Pourcentage de valeurs manquantes =  3.99 %




Pourcentage de valeurs manquantes par variable/colonne =
 Élément
Pays                                                       0
Disponibilité alimentaire en quantité (kg/personne/an)     0
Disponibilité intérieure                                   2
Importations - Quantité                                    2
Variation de stock                                         3
Production                                                 4
Exportations - Quantité                                   37
dtype: int64


---------------------------------------
2-Table population :
Nombre de valeurs manquantes = 0
Nombre de cellulles totales = 472
Pourcentage de valeurs manquantes =  0.0 %




Pourcentage de valeurs manquantes par variable/colonne =
 Pays          0
Population    0
dtype: int64


---------------------------------------
3-Table PIB Indice 

##<font color='purple'>Affichage des valeurs manquantes et affectation de valeurs</font>

In [ ]:
#affiche le df avec au moins une valeur manquante par ligne
df_dispovol_null = df_dispovol[df_dispovol.isnull().any(axis=1)]
df_dispovol_null

Élément                                   Pays  \
0                                  Afghanistan   
13                                     Bahamas   
14                                  Bangladesh   
23                                Burkina Faso   
26                                  Cabo Verde   
27                                    Cambodge   
40                                        Cuba   
43                                    Djibouti   
54                                      Gambie   
56                                     Grenade   
59                                      Guinée   
60                               Guinée-Bissau   
63                                       Haïti   
80                                    Kiribati   
89                                  Madagascar   
91                                      Malawi   
92                                    Maldives   
93                                        Mali   
97                                  Mauritanie   
99                                    Mongolie   
101                                 Mozambique   
108                         Nouvelle-Calédonie   
110                                      Népal   
113                                Ouzbékistan   
122                                      Pérou   
126                  République centrafricaine   
130      République démocratique populaire lao   
132                République-Unie de Tanzanie   
134            Saint-Vincent-et-les Grenadines   
135                               Sainte-Lucie   
137                       Sao Tomé-et-Principe   
142                                     Soudan   
148                                Tadjikistan   
152                                Timor-Leste   
156                               Turkménistan   
160                                    Vanuatu   
165                                   Zimbabwe   
170                                   Éthiopie   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
13                                                   43.17        
14                                                    1.50        
23                                                    2.27        
26                                                   17.62        
27                                                    2.34        
40                                                   23.72        
43                                                    2.68        
54                                                    3.53        
56                                                   45.70        
59                                                    4.08        
60                                                    2.16        
63                                                    8.91        
80                                                   17.98        
89                                                    2.87        
91                                                    5.17        
92                                                   13.50        
93                                                    2.83        
97                                                    5.14        
99                                                    2.77        
101                                                   3.59        
108                                                  38.71        
110                                                   1.99        
113                                                   1.96        
122                                                  13.47        
126                                                   2.29        
130                                                  10.91        
132                                                   1.88        
134                                                  72.31        
135                                    

In [ ]:
df_dispovol_null['Exportations - Quantité']=df_dispovol_null['Production']+df_dispovol_null['Importations - Quantité']+df_dispovol_null['Variation de stock']-df_dispovol_null['Disponibilité intérieure']
df_dispovol_null

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


Élément                                   Pays  \
0                                  Afghanistan   
13                                     Bahamas   
14                                  Bangladesh   
23                                Burkina Faso   
26                                  Cabo Verde   
27                                    Cambodge   
40                                        Cuba   
43                                    Djibouti   
54                                      Gambie   
56                                     Grenade   
59                                      Guinée   
60                               Guinée-Bissau   
63                                       Haïti   
80                                    Kiribati   
89                                  Madagascar   
91                                      Malawi   
92                                    Maldives   
93                                        Mali   
97                                  Mauritanie   
99                                    Mongolie   
101                                 Mozambique   
108                         Nouvelle-Calédonie   
110                                      Népal   
113                                Ouzbékistan   
122                                      Pérou   
126                  République centrafricaine   
130      République démocratique populaire lao   
132                République-Unie de Tanzanie   
134            Saint-Vincent-et-les Grenadines   
135                               Sainte-Lucie   
137                       Sao Tomé-et-Principe   
142                                     Soudan   
148                                Tadjikistan   
152                                Timor-Leste   
156                               Turkménistan   
160                                    Vanuatu   
165                                   Zimbabwe   
170                                   Éthiopie   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
13                                                   43.17        
14                                                    1.50        
23                                                    2.27        
26                                                   17.62        
27                                                    2.34        
40                                                   23.72        
43                                                    2.68        
54                                                    3.53        
56                                                   45.70        
59                                                    4.08        
60                                                    2.16        
63                                                    8.91        
80                                                   17.98        
89                                                    2.87        
91                                                    5.17        
92                                                   13.50        
93                                                    2.83        
97                                                    5.14        
99                                                    2.77        
101                                                   3.59        
108                                                  38.71        
110                                                   1.99        
113                                                   1.96        
122                                                  13.47        
126                                                   2.29        
130                                                  10.91        
132                                                   1.88        
134                                                  72.31        
135                                    

In [ ]:
df_dispovol_null[df_dispovol_null.isnull().any(axis=1)]

Élément                                   Pays  \
43                                    Djibouti   
92                                    Maldives   
113                                Ouzbékistan   
122                                      Pérou   
130      République démocratique populaire lao   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
43                                                    2.68        
92                                                   13.50        
113                                                   1.96        
122                                                  13.47        
130                                                  10.91        

Élément  Disponibilité intérieure  Exportations - Quantité  \
43                            3.0                      NaN   
92                           12.0                      NaN   
113                           NaN                      NaN   
122                        1523.0                      NaN   
130                           NaN                      NaN   

Élément  Importations - Quantité  Production  Variation de stock  
43                           3.0         NaN                 0.0  
92                          12.0         NaN                 0.0  
113                          NaN         NaN                 NaN  
122                         60.0      1465.0                 NaN  
130                          NaN         NaN                 NaN

In [ ]:
df_dispovol_null[df_dispovol_null.isnull().any(axis=1)]

Élément                                   Pays  \
43                                    Djibouti   
92                                    Maldives   
113                                Ouzbékistan   
122                                      Pérou   
130      République démocratique populaire lao   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
43                                                    2.68        
92                                                   13.50        
113                                                   1.96        
122                                                  13.47        
130                                                  10.91        

Élément  Disponibilité intérieure  Exportations - Quantité  \
43                            3.0                      NaN   
92                           12.0                      NaN   
113                           NaN                      NaN   
122                        1523.0                      NaN   
130                           NaN                      NaN   

Élément  Importations - Quantité  Production  Variation de stock  
43                           3.0         NaN                 0.0  
92                          12.0         NaN                 0.0  
113                          NaN         NaN                 NaN  
122                         60.0      1465.0                 NaN  
130                          NaN         NaN                 NaN

In [ ]:
#Affectation des valeurs manquantes par les valeurs trouvées dans mes recherches
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Pérou','Exportations - Quantité'] = 1
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Pérou','Importations - Quantité'] = 60
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Pérou','Variation de stock'] = 0
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Ouzbékistan','Importations - Quantité'] = 8
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Ouzbékistan','Disponibilité intérieure'] = 67
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Ouzbékistan','Production'] = 58
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Ouzbékistan','Variation de stock'] = -2
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Ouzbékistan','Exportations - Quantité'] = 0
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Djibouti','Importations - Quantité'] = 3
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'Maldives','Importations - Quantité'] = 12
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'République démocratique populaire lao','Importations - Quantité'] = 5
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'République démocratique populaire lao','Production'] = 33
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'République démocratique populaire lao','Variation de stock'] = 0
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'République démocratique populaire lao','Disponibilité intérieure'] = 38
df_dispovol_null.loc[df_dispovol_null['Pays'] == 'République démocratique populaire lao','Exportations - Quantité'] = 0

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


In [ ]:
df_dispovol_null[df_dispovol_null.isnull().any(axis=1)]

Élément      Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
43       Djibouti                                               2.68        
92       Maldives                                              13.50        

Élément  Disponibilité intérieure  Exportations - Quantité  \
43                            3.0                      NaN   
92                           12.0                      NaN   

Élément  Importations - Quantité  Production  Variation de stock  
43                           3.0         NaN                 0.0  
92                          12.0         NaN                 0.0

In [ ]:
#Il reste toujours 4 valeurs manquantes pour l'exportation et la production des Maldives et Djibouti 
#que je vais remplacer par 0
df_dispovol_null=df_dispovol_null.fillna(0)
df_dispovol_null.head()

Élément          Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0         Afghanistan                                               1.53        
13            Bahamas                                              43.17        
14         Bangladesh                                               1.50        
23       Burkina Faso                                               2.27        
26         Cabo Verde                                              17.62        

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                            57.0                      0.0   
13                           26.0                      8.0   
14                          250.0                     -1.0   
23                           46.0                      0.0   
26                           10.0                      7.0   

Élément  Importations - Quantité  Production  Variation de stock  
0                           29.0        28.0                 0.0  
13                          24.0         6.0                 4.0  
14                           0.0       249.0                 0.0  
23                           0.0        46.0                 0.0  
26                          12.0         1.0                 4.0

In [ ]:
df_dispovol_null[df_dispovol_null.isnull().any(axis=1)]

Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock]
Index: []

In [ ]:
df_dispovol.head()

Élément            Pays  \
0           Afghanistan   
1        Afrique du Sud   
2               Albanie   
3               Algérie   
4             Allemagne   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
1                                                    35.69        
2                                                    16.36        
3                                                     6.38        
4                                                    19.47        

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                            57.0                      NaN   
1                          2118.0                     63.0   
2                            47.0                      0.0   
3                           277.0                      0.0   
4                          1739.0                    646.0   

Élément  Importations - Quantité  Production  Variation de stock  
0                           29.0        28.0                 0.0  
1                          514.0      1667.0                 0.0  
2                           38.0        13.0                 4.0  
3                            2.0       275.0                 0.0  
4                          842.0      1514.0               -29.0

In [ ]:
df_dispovol=df_dispovol.dropna()

In [ ]:
df_dispovol=df_dispovol.append(df_dispovol_null)

In [ ]:
df_dispovol[df_dispovol.isnull().any(axis=1)]

Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock]
Index: []

In [ ]:
df_dispovol.sort_values(by='Pays', ascending=True).head()

Élément            Pays  \
0           Afghanistan   
1        Afrique du Sud   
2               Albanie   
3               Algérie   
4             Allemagne   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
1                                                    35.69        
2                                                    16.36        
3                                                     6.38        
4                                                    19.47        

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                            57.0                      0.0   
1                          2118.0                     63.0   
2                            47.0                      0.0   
3                           277.0                      0.0   
4                          1739.0                    646.0   

Élément  Importations - Quantité  Production  Variation de stock  
0                           29.0        28.0                 0.0  
1                          514.0      1667.0                 0.0  
2                           38.0        13.0                 4.0  
3                            2.0       275.0                 0.0  
4                          842.0      1514.0               -29.0

In [ ]:
df_pop2017[df_pop2017.isnull().any(axis=1)]

Empty DataFrame
Columns: [Pays, Population]
Index: []

Pas de valeur manquante dans la table population

In [ ]:
df_pibind[df_pibind['PIB par habitant, ($ PPA internationaux constants de 2011)'].isna()]

Produit                                        Pays  \
5                                           Andorre   
40                        Chine, Taiwan Province de   
47                                             Cuba   
64                                        Groenland   
141                       République arabe syrienne   
148      République populaire démocratique de Corée   
154                               Samoa américaines   
164                                   Soudan du Sud   
185          Venezuela (République bolivarienne du)   
187                                           Yémen   
193                                        Érythrée   

Produit  PIB par habitant, ($ PPA internationaux constants de 2011)  \
5                                                      NaN            
40                                                     NaN            
47                                                     NaN            
64                                                     NaN            
141                                                    NaN            
148                                                    NaN            
154                                                    NaN            
164                                                    NaN            
185                                                    NaN            
187                                                    NaN            
193                                                    NaN            

Produit  Stabilité politique et absence de violence/terrorisme (indice)  
5                                                     1.42               
40                                                    0.86               
47                                                    0.65               
64                                                    1.92               
141                                                  -2.62               
148                                                  -0.50               
154                                                   1.22               
164                                                  -2.45               
185                                                  -1.25               
187                                                  -2.94               
193                                                  -0.73

In [ ]:
#Affectation des valeurs manquantes par les valeurs trouvées dans mes recherches
df_pibind.loc[df_pibind['Pays'] == 'Andorre','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 56352.2
df_pibind.loc[df_pibind['Pays'] == 'Chine, Taiwan Province de','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 48500.8
df_pibind.loc[df_pibind['Pays'] == 'Samoa américaines','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 5668
df_pibind.loc[df_pibind['Pays'] == 'Soudan du Sud','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 888.8
df_pibind.loc[df_pibind['Pays'] == 'Yémen','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 2150
df_pibind.loc[df_pibind['Pays'] == 'Érythrée','PIB par habitant, ($ PPA internationaux constants de 2011)'] = 1537.5

In [ ]:
#Il reste toujours des valeurs manquantes pour 6 pays pour les 2 variables PIB par habitant et l'indice de Stabilité politique 
#que je vais remplacer par 0
df_pibind=df_pibind.fillna(0)
df_pibind.head()

Produit            Pays  \
0           Afghanistan   
1        Afrique du Sud   
2               Albanie   
3               Algérie   
4             Allemagne   

Produit  PIB par habitant, ($ PPA internationaux constants de 2011)  \
0                                                   2058.4            
1                                                  12701.3            
2                                                  12771.0            
3                                                  11737.4            
4                                                  52952.9            

Produit  Stabilité politique et absence de violence/terrorisme (indice)  
0                                                    -2.80               
1                                                    -0.28               
2                                                     0.38               
3                                                    -0.92               
4                                                     0.59

In [ ]:
df_pibind[df_pibind.isnull().any(axis=1)]

Empty DataFrame
Columns: [Pays, PIB par habitant, ($ PPA internationaux constants de 2011), Stabilité politique et absence de violence/terrorisme (indice)]
Index: []

# <a name="C6"><font color='LightSeaGreen'>Unicité de la clé primaire</font></a>

In [ ]:
#Creation d'une fonction qui teste l'unicité d'une colonne
def cle_primaire(df,col):
    if len(df)==len(df[col].unique()):
        print('La variable', col, 'est une clé primaire')
    else: 
        print('La variable', col, 'n\'est pas une clé primaire')

In [ ]:
cle_primaire(df_dispovol,'Pays')
cle_primaire(df_pop2017,'Pays')
cle_primaire(df_pibind,'Pays')

La variable Pays est une clé primaire
La variable Pays est une clé primaire
La variable Pays est une clé primaire


# <a name="C7"><font color='LightSeaGreen'>Analyse des doublons</font></a>

In [ ]:
#Fonction qui detecte les doublons
def detect_doublons(df):
    doub=df[df[df.columns].duplicated(subset=None, keep='first')]
    print('------------------') 
    print('La table contient', len(doub), 'doublons.')
    print(doub.shape)
    print('\n') 
    print(doub)
    print('\n')

In [ ]:
print('------------------') 
print('Table disponibilité alimentaire')
detect_doublons(df_dispovol)
print('------------------') 
print('Table population')
detect_doublons(df_pop2017)
print('------------------') 
print('Table pib indice')
detect_doublons(df_pibind)

------------------
Table disponibilité alimentaire
------------------
La table contient 0 doublons.
(0, 7)


Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock]
Index: []


------------------
Table population
------------------
La table contient 0 doublons.
(0, 2)


Empty DataFrame
Columns: [Pays, Population]
Index: []


------------------
Table pib indice
------------------
La table contient 0 doublons.
(0, 3)


Empty DataFrame
Columns: [Pays, PIB par habitant, ($ PPA internationaux constants de 2011), Stabilité politique et absence de violence/terrorisme (indice)]
Index: []




# <a name="C8"><font color='LightSeaGreen'>Jointure des tables</font></a>

In [ ]:
#Jointure externe complète entre disponibilité alimentaire et population
df_dispop=pd.merge(df_dispovol, df_pop2017, left_on='Pays', right_on='Pays', how='outer', indicator=True)
df_dispop.head()

Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0  Afrique du Sud                                              35.69        
1         Albanie                                              16.36        
2         Algérie                                               6.38        
3       Allemagne                                              19.47        
4          Angola                                              10.56        

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                    2118.0                     63.0                    514.0   
1                      47.0                      0.0                     38.0   
2                     277.0                      0.0                      2.0   
3                    1739.0                    646.0                    842.0   
4                     319.0                      0.0                    277.0   

   Production  Variation de stock  Population _merge  
0      1667.0                 0.0  57009756.0   both  
1        13.0                 4.0   2884169.0   both  
2       275.0                 0.0  41389189.0   both  
3      1514.0               -29.0  82658409.0   both  
4        42.0                 0.0  29816766.0   both

In [ ]:
df_dispop[df_dispop.isnull().any(axis=1)]

Pays  \
172                      Andorre   
173                     Anguilla   
174  Antilles néerlandaises (ex)   
175                        Aruba   
176                      Bahreïn   
..                           ...   
231                      Somalie   
232                Soudan du Sud   
233                     Tokélaou   
234                        Tonga   
235                       Tuvalu   

     Disponibilité alimentaire en quantité (kg/personne/an)  \
172                                                NaN        
173                                                NaN        
174                                                NaN        
175                                                NaN        
176                                                NaN        
..                                                 ...        
231                                                NaN        
232                                                NaN        
233                                                NaN        
234                                                NaN        
235                                                NaN        

     Disponibilité intérieure  Exportations - Quantité  \
172                       NaN                      NaN   
173                       NaN                      NaN   
174                       NaN                      NaN   
175                       NaN                      NaN   
176                       NaN                      NaN   
..                        ...                      ...   
231                       NaN                      NaN   
232                       NaN                      NaN   
233                       NaN                      NaN   
234                       NaN                      NaN   
235                       NaN                      NaN   

     Importations - Quantité  Production  Variation de stock  Population  \
172                      NaN         NaN                 NaN     77001.0   
173                      NaN         NaN                 NaN     14584.0   
174                      NaN         NaN                 NaN    275186.0   
175                      NaN         NaN                 NaN    105366.0   
176                      NaN         NaN                 NaN   1494076.0   
..                       ...         ...                 ...         ...   
231                      NaN         NaN                 NaN  14589179.0   
232                      NaN         NaN                 NaN  10910763.0   
233                      NaN         NaN                 NaN      1300.0   
234                      NaN         NaN                 NaN    101998.0   
235                      NaN         NaN                 NaN     11370.0   

         _merge  
172  right_only  
173  right_only  
174  right_only  
175  right_only  
176  right_only  
..          ...  
231  right_only  
232  right_only  
233  right_only  
234  right_only  
235  right_only  

[64 rows x 9 columns]

In [ ]:
#Suppression des pays qui n'ont aucune information dans la table disponibilité alimentaire
df_dispop=df_dispop.drop(df_dispop[df_dispop['_merge']=='right_only'].index)
df_dispop.head()

Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0  Afrique du Sud                                              35.69        
1         Albanie                                              16.36        
2         Algérie                                               6.38        
3       Allemagne                                              19.47        
4          Angola                                              10.56        

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                    2118.0                     63.0                    514.0   
1                      47.0                      0.0                     38.0   
2                     277.0                      0.0                      2.0   
3                    1739.0                    646.0                    842.0   
4                     319.0                      0.0                    277.0   

   Production  Variation de stock  Population _merge  
0      1667.0                 0.0  57009756.0   both  
1        13.0                 4.0   2884169.0   both  
2       275.0                 0.0  41389189.0   both  
3      1514.0               -29.0  82658409.0   both  
4        42.0                 0.0  29816766.0   both

In [ ]:
df_dispop[df_dispop.isnull().any(axis=1)]

Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock, Population, _merge]
Index: []

In [ ]:
#Jointure externe complète entre dispop et la table du PIB et de la stabilité politique
df_final=pd.merge(df_dispop, df_pibind, left_on='Pays', right_on='Pays', how='outer')
df_final

Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0    Afrique du Sud                                              35.69        
1           Albanie                                              16.36        
2           Algérie                                               6.38        
3         Allemagne                                              19.47        
4            Angola                                              10.56        
..              ...                                                ...        
196   Soudan du Sud                                                NaN        
197           Tonga                                                NaN        
198          Tuvalu                                                NaN        
199        Érythrée                                                NaN        
200   Îles Marshall                                                NaN        

     Disponibilité intérieure  Exportations - Quantité  \
0                      2118.0                     63.0   
1                        47.0                      0.0   
2                       277.0                      0.0   
3                      1739.0                    646.0   
4                       319.0                      0.0   
..                        ...                      ...   
196                       NaN                      NaN   
197                       NaN                      NaN   
198                       NaN                      NaN   
199                       NaN                      NaN   
200                       NaN                      NaN   

     Importations - Quantité  Production  Variation de stock  Population  \
0                      514.0      1667.0                 0.0  57009756.0   
1                       38.0        13.0                 4.0   2884169.0   
2                        2.0       275.0                 0.0  41389189.0   
3                      842.0      1514.0               -29.0  82658409.0   
4                      277.0        42.0                 0.0  29816766.0   
..                       ...         ...                 ...         ...   
196                      NaN         NaN                 NaN         NaN   
197                      NaN         NaN                 NaN         NaN   
198                      NaN         NaN                 NaN         NaN   
199                      NaN         NaN                 NaN         NaN   
200                      NaN         NaN                 NaN         NaN   

    _merge  PIB par habitant, ($ PPA internationaux constants de 2011)  \
0     both                                            12701.3            
1     both                                            12771.0            
2     both                                            11737.4            
3     both                                            52952.9            
4     both                                             7310.9            
..     ...                                                ...            
196    NaN                                              888.8            
197    NaN                                             6467.1            
198    NaN                                             3896.7            
199    NaN                                             1537.5            
200    NaN                                             3706.3            

     Stabilité politique et absence de violence/terrorisme (indice)  
0                                                -0.28               
1                                                 0.38               
2                                                -0.92               
3                                                 0.59               
4                                                -0.33               
..                                                 ...               
196                                              -2.45               
197    

In [ ]:
df_final[df_final.isnull().any(axis=1)].head()

Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock, Population, PIB par habitant, ($ PPA internationaux constants de 2011), Stabilité politique et absence de violence/terrorisme (indice)]
Index: []

In [ ]:
#Suppression des pays qui n'ont aucune information dans la table disponibilité alimentaire
df_final=df_final.dropna()
df_final

Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0    Afrique du Sud                                              35.69        
1           Albanie                                              16.36        
2           Algérie                                               6.38        
3         Allemagne                                              19.47        
4            Angola                                              10.56        
..              ...                                                ...        
167     Timor-Leste                                               4.24        
168    Turkménistan                                               4.52        
169         Vanuatu                                              11.66        
170        Zimbabwe                                               4.68        
171        Éthiopie                                               0.13        

     Disponibilité intérieure  Exportations - Quantité  \
0                      2118.0                     63.0   
1                        47.0                      0.0   
2                       277.0                      0.0   
3                      1739.0                    646.0   
4                       319.0                      0.0   
..                        ...                      ...   
167                       5.0                     14.0   
168                      27.0                      4.0   
169                       3.0                      3.0   
170                      76.0                     -1.0   
171                      14.0                      1.0   

     Importations - Quantité  Production  Variation de stock   Population  \
0                      514.0      1667.0                 0.0   57009756.0   
1                       38.0        13.0                 4.0    2884169.0   
2                        2.0       275.0                 0.0   41389189.0   
3                      842.0      1514.0               -29.0   82658409.0   
4                      277.0        42.0                 0.0   29816766.0   
..                       ...         ...                 ...          ...   
167                     11.0         1.0                 7.0    1243258.0   
168                      9.0        20.0                 2.0    5757667.0   
169                      4.0         1.0                 1.0     285510.0   
170                      6.0        69.0                 0.0   14236595.0   
171                      1.0        14.0                 0.0  106399924.0   

    _merge  PIB par habitant, ($ PPA internationaux constants de 2011)  \
0     both                                            12701.3            
1     both                                            12771.0            
2     both                                            11737.4            
3     both                                            52952.9            
4     both                                             7310.9            
..     ...                                                ...            
167   both                                             3145.5            
168   both                                            14205.0            
169   both                                             3081.8            
170   both                                             3274.6            
171   both                                             2021.6            

     Stabilité politique et absence de violence/terrorisme (indice)  
0                                                -0.28               
1                                                 0.38               
2                                                -0.92               
3                                                 0.59               
4                                                -0.33               
..                                                 ...               
167                                               0.07           

In [ ]:
#suppression de la colonne _merge
df_final=df_final.drop(['_merge'], axis = 1)

In [ ]:
df_final.head()

Pays  Disponibilité alimentaire en quantité (kg/personne/an)  \
0  Afrique du Sud                                              35.69        
1         Albanie                                              16.36        
2         Algérie                                               6.38        
3       Allemagne                                              19.47        
4          Angola                                              10.56        

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                    2118.0                     63.0                    514.0   
1                      47.0                      0.0                     38.0   
2                     277.0                      0.0                      2.0   
3                    1739.0                    646.0                    842.0   
4                     319.0                      0.0                    277.0   

   Production  Variation de stock  Population  \
0      1667.0                 0.0  57009756.0   
1        13.0                 4.0   2884169.0   
2       275.0                 0.0  41389189.0   
3      1514.0               -29.0  82658409.0   
4        42.0                 0.0  29816766.0   

   PIB par habitant, ($ PPA internationaux constants de 2011)  \
0                                            12701.3            
1                                            12771.0            
2                                            11737.4            
3                                            52952.9            
4                                             7310.9            

   Stabilité politique et absence de violence/terrorisme (indice)  
0                                              -0.28               
1                                               0.38               
2                                              -0.92               
3                                               0.59               
4                                              -0.33

In [ ]:
#récupère les lignes où l'exportation est négative
df_final[df_final['Exportations - Quantité']<0]

In [ ]:
#remplace les valeurs négatives de l'exportation par 0
df_final['Exportations - Quantité']=df_final['Exportations - Quantité'].apply(lambda x: 0 if x < 0 else x)

In [ ]:
df_final[df_final['Exportations - Quantité']<0]

Empty DataFrame
Columns: [Pays, Disponibilité alimentaire en quantité (kg/personne/an), Disponibilité intérieure, Exportations - Quantité, Importations - Quantité, Production, Variation de stock, Population, PIB par habitant, ($ PPA internationaux constants de 2011), Stabilité politique et absence de violence/terrorisme (indice)]
Index: []

# <a name="C9"><font color='LightSeaGreen'>Indicateurs choisis</font></a>

1.   Disponibilité d'aliments dans chaque pays en kg/personne/an 
2.   Importations (Milliers de tonnes)
3.   Production (Milliers de tonnes)
4.   Population : si le nombre d'habitants est grand, la probabilité que les habitants consomment du poulet est plus grande, alors il serait plus intéressant d'y envoyer notre poulet 
5.   PIB/Habitant ($) : si le PIB d'un pays est fort, cela indique un niveau economique élevé du pays en moyenne, donc la consommation de poulet pourrait y être plus élevée 
6.   Indice de stabilité politique 
(-2.5 : faible stabilité politique (min) / 2.5 : forte stabilité politique (max)) : une instabilité politique dans un pays empêcherait l'importation de poulet 

